src/bun.js/bindings/bindings.cpp

```diff
@@ -129,23 +129,51 @@ using namespace JSC;
 
 using namespace WebCore;
 
+typedef uint8_t ExpectFlags;
+
+// Note: keep this in sync with Expect.Flags implementation in zig (at expect.zig)
+static const int FLAG_PROMISE_RESOLVES = (1 << 0);
+static const int FLAG_PROMISE_REJECTS = (1 << 1);
+static const int FLAG_NOT = (1 << 2);
+
+extern "C" bool Expect_readFlagsAndProcessPromise(JSC__JSValue instanceValue, JSC__JSGlobalObject* globalObject, ExpectFlags* flags, JSC__JSValue* value);
+extern "C" bool ExpectCustomAsymmetricMatcher__execute(void* self, JSC__JSValue thisValue, JSC__JSGlobalObject* globalObject, JSC__JSValue leftValue);
+
 enum class AsymmetricMatcherResult : uint8_t {
     PASS,
     FAIL,
     NOT_MATCHER,
 };
 
-AsymmetricMatcherResult matchAsymmetricMatcher(JSGlobalObject* globalObject, JSCell* matcherPropCell, JSValue otherProp, ThrowScope* throwScope)
+template<class T>
+bool readFlagsAndProcessPromise(JSValue& instanceValue, T* instance, ExpectFlags& flags, JSGlobalObject* globalObject, JSValue& value)
+{
```

Comment:
is the template necessary? `instance` doesn't seem to be used

Reply:
true! it was a leftover of when the flags where stored as JSValue